Lib/idlelib/idle_test/test_debugger.py

```diff
@@ -2,9 +2,11 @@
 
 from idlelib import debugger
 import unittest
+from unittest import mock
 from test.support import requires
-requires('gui')
+# requires('gui')
```

Comment:
This is required to guard tkinter.Tk() calls, which fail on buildbots without graphics systems.  If there are non-gui test classes, like those you added, then it should go into setUpClass, before the call, so that the non-gui classes can run on such systems.  In this case, move it into NameSpaceTest.setUpClass at the top.  Any other gui test classes will get the same.

Reply:
I forgot to uncomment this when checking in. It's the only way I can get the tests to run on Mac OS

Reply:
You should not have to do that.  On my MacBook with Mohave, installed 3.7 runs test_idle just fine.